helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/backports/SpecificRecordBaseExt.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2025 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.compatibility.backports;
+
+import com.linkedin.avroutil1.compatibility.CustomDecoder;
+import org.apache.avro.specific.SpecificRecordBase;
+
+import java.io.IOException;
+
+/**
+ * Extension of {@link SpecificRecordBase} that allows for custom decoding using a custom decoder.
+ */
+public abstract class SpecificRecordBaseExt extends SpecificRecordBase {
```

Comment:
Great idea using this to cleanly extend generated classes with `customDecode(CustomDecoder)` while keeping full compatibility (`instanceof SpecificRecordBase` still holds).
👏